modin/core/dataframe/base/partitioning/axis_partition.py

```diff
@@ -105,7 +116,23 @@ def _wrap_partitions(self, partitions: list) -> list:
         assert self.partition_type is not None
         assert self.instance_type is not None  # type: ignore
 
-        return [self.partition_type(obj) for obj in partitions]
+        if extract_metadata is None:
+            # If `_PARTITIONS_METADATA_LEN == 0` then the execution doesn't support metadata
+            # and thus we should never try extracting it, otherwise assuming that the common
+            # approach of always passing the metadata is used.
+            extract_metadata = bool(self._PARTITIONS_METADATA_LEN)
+
+        if extract_metadata:
```

Comment:
Would be nice to add a comment explaining what's going on here (i.e. that if `extract_metadata` is `True`, then partitions is actually a list of [data0, metadata0, metadata1, ..., data1, metadata0, metadata1, ...], and so we need to zip together the data and metadata.

Reply:
Added the comment